### DISCUSSION TITLE
uniform drift variability log-likelihood

### DISCUSSION THREAD

**Discussion by ntardiff** [mwshinn/PyDDM#66] [Ask for help] [answered]
You note the in cookbook entry for uniform drift variability that the log-likelihood will be inaccurate as a result of needing to duplicate the sample. Do you have any sense of whether dividing the log-likelihood by the uniform distribution resolution would give a reasonable approximation to the log-likelihood that would be comparable to fits without drift variability?

Also, to generate predictions from the model, would you average over the discretized drift-rate distribution?

Thanks,
Nathan

**Comment by mwshinn** [comment, marked as answer]
Hi Nathan,

Unfortunately it's a bit more complicated than that.  If you think of this as a mixture model, then in the simplest case of a discretisation of 2, you have $p(x) = p_1(x)/2 + p_2(x)/2$.  So then the equation for likelihood becomes

$$
L = \prod_i p(x_i)
$$

$$
L = 1/2^n \prod_i (p_1(x_i) + p_2(x_i))
$$

$$
log(L) = -n log(2) + \sum_i log(p_1(x_i) + p_2(x_i))
$$

Not sure if there is some shortcut you can derive, but you might have to calculate likelihood point by point for this case since I don't see a way to mathematically isolate the terms $\sum_i log(p_k(x_i))$.

If you are up for doing the likelihood calculation manually, you could alternatively do it using the curve from "solve_partial_conditions" for your discretisation as a condition, since that is equal to the sum $(p_1 + p_2)/2$.

To generate predictions, first sample a value for the drift rate, and then sample from the full model for that drift rate.  Or just use "solve_partial_conditions" and sample from the resulting distribution (using Solution.resample())

**Reply by ntardiff** [reply]
Thanks, Max! I had a feeling it wasn't that simple. Solving and then doing the calculation manually based on the predicted choice probabilities wouldn't be that bad...

Just to be clear, when I said predictions I meant solving the model, not generating samples. Using solve_all_conditions and then averaging over the $p_k$ from each drift rate would be equivalent to using solve_partial_conditions and not specifying the driftnum variable, correct?

**Reply by mwshinn** [reply]
Yes, but you need a sample to calculate the likelihood.  So to calculate the likelihood, you would use solve_partial_conditions (which, if you pass it all the driftnum variables you care about, will give you a weighted average of all of the pdfs, equivalent to calculating it manually from solve_all_conditions) and then take the resulting average pdf and find the value at each RT from the sample for correct or error.  Then take the log of each and sum them (i.e. apply the definition of log likelihood).

**Reply by ntardiff** [reply]
Got it. Thanks!
